website/blog/2020-01-10-cobalt.md

```diff
@@ -0,0 +1,953 @@
+and consider suggesting other refactors that are essential in your development
+workflow.
+
+### Rename symbol
+
+Thanks to the implementation of the `Go to implementations` feature, we were
+able to work on a proper rename for any workspace symbol that will rename all
+occurrences of a symbol even if that symbol is a part of a class hierarchy.
+
+![rename](https://i.imgur.com/QHaJCr0.gif)
+
+There are a couple of cases in which rename might behave differently based on
+the particular symbol being renamed:
+
+- renaming an overriden symbol will rename all occurrences for both parent and
+  all children overriding that symbol
+- java symbols are not renamed
+- renaming "hashCode", "unapply", "unary\_!", "!" is not supported, since they
+  change the semantics
+- methods starting with `:` can only be renamed to ones with the same first
+  character to make sure code compiles afterwards
+- renaming apply will change all occurrences with just `()` to a full name
+  `.newname()`
+- if renaming a class with the same name as the file, the file's name will also
+  be renamed
+- companion objects will also be renamed along with their companion classes
+
+### Import missing symbol
+
+Another refactor that is commonly requested is the ability to fix a compilation
+error by auto-importing a symbol present on the classpath.
+
+This was previously possible in Metals only during completions, but we were
+lacking what LSP calls a code action.
+
+Code actions are context-sensitive actions that are proposed to the user in
+order to fix an error or to perform a relevant refactor.
+
+Thanks to the work of @gabro we now have a basic architecture in place for
+supporting code actions and we were able to implement this feature.
+
+![missing-symbol](https://user-images.githubusercontent.com/691940/68937367-8839bf80-079c-11ea-8401-4927f6531d24.gif)
+
+## Worksheet support
+
+Thanks to great work by @olafurpg and some additional refinements by
+@alekseiAlefirov, worksheets are now available in all supported editors.
+
+Worksheets are a way to evaluate code in real time to test and prototype any new
+features in a simple way. To create a worksheet all that is needed is to create
+a file with the extension `worksheet.sc` and Metals will take care of the rest.
+
+There are two ways worksheets are implemented for different editors. One is with
+an additional extension to LSP called `Decoration extension` and it's used by
+Visual Studio Code and coc-metals.
+
+Visual Studio Code:
+
+![worksheet](https://user-images.githubusercontent.com/1408093/68093266-b0b3e680-fe8b-11e9-92dd-0165f169966f.gif)
+
+The other way worksheets are implemented for all other editors that don't
+support the `Decoration extension` is using workspace/applyEdits and comments
+with hover for larger results.
+
+Sublime:
+
+![worksheet2](https://user-images.githubusercontent.com/10850363/70265661-29fc6d00-179b-11ea-842e-935045e27f96.gif)
+
+## Build tools
+
+### Experimental build tool support for Pants
```

Comment:
This section did not incorporate yet the full suggestion from  https://github.com/scalameta/metals/pull/1278#discussion_r365410510

> Thanks to the recent work by [@olafurpg](https://github.com/olafurpg), there's now experimental support for the Pants build tool. The Pants integration is still evolving so it's not yet documented. The end goal is to support Pants in the same way as other build tools like sbt, Maven and Gradle. 


We don't need to say more about Pants for now.

Reply:
Missed it! Sorry - added that paragraph and removed the previous version